server/backend/src/main/java/eu/bbmri_eric/quality/server/user/UserService.java

```diff
@@ -3,6 +3,15 @@
 public interface UserService {
   UserDTO createUser(UserCreateDTO userCreateDTO);
 
+  /**
+   * Find a user by subject ID, or create a new user if not found.
+   *
+   * @param subjectId the OIDC subject ID
+   * @param username the username to use if creating a new user
+   * @return the found or created user
+   */
+  UserDTO findOrCreateUserBySubjectId(String subjectId, String username);
```

Comment:
IMHO this is a very specific method to be in an API. Why not have 2 general purpose methods findBySubjectID and create?